pkg/server/postgres/postgres.go

```diff
@@ -176,3 +187,92 @@ func (p *PostgresPlugin) DeleteDatabase(db *atlas.Database) error {
 	// delete database
 	return nil
 }
+
+func checkUsers(db *atlas.Database, sqldb *sql.DB) error {
+	//// check if users exist; verify their passwords and roles
+	dbUserMap := make(map[string]string)
+	var u string
+	//getting all users from the database
+	rows, err := sqldb.Query(`SELECT u.usename FROM pg_catalog.pg_user u`)
+	if err != nil {
+		return err
+	}
+	for rows.Next() {
+		if err := rows.Scan(&u); err != nil {
+			return err
+		}
+		dbUserMap[u] = "delete me"
+	}
+	glog.V(4).Info("Usersexist in database are", dbUserMap)
+	users := db.Spec.Users
+	glog.V(4).Info("Users from yaml are", users)
+
+	for _, user := range users {
+		if _, ok := dbUserMap[user.Name]; !ok {
+			err = createuser(user.Name, user.Role, user.Password, sqldb)
+			if err != nil {
+				return err
+			}
+		} else {
+			dbUserMap[user.Name] = "dont delete"
+			if user.Role == "admin" {
+				//roles now assigned to admin is createdb, createrole
+				_, err = sqldb.Exec(fmt.Sprintf(`alter user "%s" with %s`, user.Name, admin_role))
+				if err != nil {
+					return err
+				}
+			} else if user.Role == "read" {
+				_, err = sqldb.Exec(fmt.Sprintf(`alter user "%s" with %s`, user.Name, read_role))
+				if err != nil {
+					return err
+				}
+			} else {
+				return fmt.Errorf("role for the user %s is not valid", user.Name)
+			}
+			//There is no straight forward ways to compare password so just updating it
+			_, err = sqldb.Exec(fmt.Sprintf(`alter user "%s" with password '%s'`, user.Name, user.Password))
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	// delete users
+	for dbUserName, dbUserVal := range dbUserMap {
+		//if not-visited delete it
+		if dbUserVal == "dont delete" && dbUserName != "postgres" {
+			glog.Info("inside delete, not visited", dbUserName)
+			err = deleteUser(dbUserName, sqldb)
```

Comment:
will it delete user created by another app with different yaml file?

Reply:
yes. that we should change.  we have to find a logic to delete only users related to specific app. for that may be we can restrict username attached with app name or db name. In that case we can validate username having app name part or not.

Reply:
for now will disable the deletion part.